.github/workflows/release-build-publish.yml

```diff
@@ -0,0 +1,256 @@
+          ACTUAL_DIRS=$(find $PACKAGE_DIR -maxdepth 1 -type d -exec basename {} \; | sed "1d") # Remove the package directory itself from the list
+
+          # Function to check for unexpected or missing files and directories
+          check_contents() {
+            local missing=0
+            # Check for expected files
+            for expected_file in $EXPECTED_FILES; do
+              echo $ACTUAL_FILES | grep -qw $expected_file || { echo "Missing expected file: $expected_file"; missing=1; }
+            done
+            
+            # Check for expected directories
+            for expected_dir in $EXPECTED_DIRS; do
+              echo $ACTUAL_DIRS | grep -qw $expected_dir || { echo "Missing expected directory: $expected_dir"; missing=1; }
+            done
+
+            # Check for unexpected files and directories
+            for actual in $ACTUAL_CONTENTS; do
+              echo $EXPECTED_FILES $EXPECTED_DIRS | grep -qw $actual || { echo "Unexpected item in package: $actual"; missing=1; }
+            done
+
+            # Exit if anything unexpected or missing
+            if [ $missing -ne 0 ]; then
+              echo "Package content verification failed."
+              exit 1
+            fi
+          }
+
+          # Execute checks
+          check_contents
+
+      - name: Store build artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: package
+          path: dist/*
+
+  publish-testpypi:
+    needs: build
+    runs-on: ubuntu-latest
+    permissions:
+        id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+
+    steps:
+      - name: Fetch build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: package
+          path: dist/
+
+      - name: Publish to TestPyPI
+        uses: pypa/gh-action-pypi-publish@e53eb8b103ffcb59469888563dc324e3c8ba6f06
+        with:
+          repository-url: https://test.pypi.org/legacy/
+          verify-metadata: true
+
+  verify-version:
+    needs: [publish-testpypi, store_published_version, extract_project_name]
+    runs-on: ubuntu-latest
+    environment: testpypi
+
+    steps:
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install package from TestPyPI
+        env:
+          CLEANLAB_PACKAGE_NAME: ${{ needs.extract_project_name.outputs.cleanlab_package_name }}
+        run: pip install -i https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ $CLEANLAB_PACKAGE_NAME
+
+      - name: Verify the published version
+        env:
+          VERSION: ${{ needs.store_published_version.outputs.version }}
+        run: |
+          TEST_IMPORT_VERSION=$(python -c "import cleanlab; print(cleanlab.__version__)")
+          if [ "$TEST_IMPORT_VERSION" != "$VERSION" ]; then
+            echo "Imported version $TEST_IMPORT_VERSION does not match tag $VERSION"
+            exit 1
+          fi
```

Comment:
This is the step where we usually fail if the installed version is older.

It's a simple matter of just delaying the job slightly longer to make sure the test index managed to refresh.